pyblish_ftrack/plugins/collect_ftrack_api.py

```diff
@@ -19,7 +19,7 @@ def process(self, context):
         context.data["ftrackSession"] = session
 
         # Collect task
-        taskid = ""
+
```

Comment:
Will this not come up with an "UnboundLocalError: local variable 'taskid' referenced before assignment"?

Reply:
task_id is only ever assigned to in this class, so nothing references it.

Reply:
But we first check whether we can get the task id from connect event https://github.com/pyblish/pyblish-ftrack/blob/ff2a1f273920f2e94588fd5f93da38b6f221d8c6/pyblish_ftrack/plugins/collect_ftrack_api.py#L30, then we fall back to env value, and lastly to empty string.

Reply:
Yes, this is the current behaviour. 

What I'm suggesting is that the env value or empty string fallback is the default taskid assignment, which is reassigned as part of a try. That would be the same behaviour?

Reply:
Additionally can't collect_ftrack_data just lookup the id of the task from the task on the context which this plugin sets? Rather than duplicating the same code...

Reply:
> That would be the same behaviour?

Sorry, yeah you are right. That would be the same behaviour. Let's do it :)

> Additionally can't collect_ftrack_data just lookup the id of the task from the task on the context which this plugin sets? Rather than duplicating the same code...

You are right, that we have duplication of code here. Initially it was to completely separate the old api from the new api workflows. But since they are both relying on the task id and they are getting it the same way, we could definitely refactor this to a single plugin that gets the task id.
We would still need the two current plugins that collects the old api data, and the new api session and task object.